test/controllers/trust.details.controller.spec.ts

```diff
@@ -255,8 +278,143 @@ describe('Trust Details controller', () => {
 
       (mapBeneficialOwnerToSession as jest.Mock).mockReturnValue([]);
 
+      await post(mockReq, mockRes, mockNext);
+
+      expect((generateTrustId as jest.Mock)).not.toBeCalled();
+
+      expect(saveTrustInApp).toBeCalledTimes(1);
+      expect(saveTrustInApp).toBeCalledWith(mockAppData, expect.objectContaining(expectTrustResult));
+
+      expect(mapBeneficialOwnerToSession).toBeCalledTimes(2);
+
+      expect(mockSetExtraData).toBeCalledWith(
+        mockReq.session,
+        expect.objectContaining({
+          ...mockAppData,
+          [BeneficialOwnerOtherKey]: [],
+          [BeneficialOwnerIndividualKey]: [],
+        }),
+      );
+
+      expect(mockNext).not.toBeCalled();
+
+      expect(mockRes.redirect).toBeCalledTimes(1);
+      expect(mockRes.redirect).toBeCalledWith(TRUST_ENTRY_URL + "/" + mockTrust2Data.trust_id + TRUST_INVOLVED_URL);
+    });
+  });
+
+  describe('POST with url params tests', () => {
+    beforeEach(() => {
+      mockReq.body = {
+        id: 'dummyId',
+        name: 'dummyName',
+        createdDay: '77',
+        createdMonth: '88',
+        createdYears: '9999',
+        beneficialOwnersIds: ['bo1id', 'bo2id'],
+      };
+    });
+
+    test('catch error when post data from page', () => {
+      const error = new Error(ANY_MESSAGE_ERROR);
+      mockGetApplicationData.mockImplementationOnce(() => {
+        throw error;
+      });
+
       post(mockReq, mockRes, mockNext);
 
+      expect(mockNext).toBeCalledTimes(1);
```

Comment:
should the post on line 326 also have the await?

Reply:
Yes, probably. Will add.

Reply:
Done.